lib/ansible/modules/cloud/google/gce_instance_template.py

```diff
@@ -0,0 +1,582 @@
+    module : AnsibleModule object
+    gce: authenticated GCE libcloud driver
+    Returns:
+        instance template information
+    """
+    name = module.params.get('name')
+    current_state = "absent"
+    changed = False
+
+    # get instance template
+    instance = None
+    try:
+        instance = gce.ex_get_instancetemplate(name)
+        current_state = "present"
+    except GoogleBaseError as err:
+        json_data = dict(msg='instance template not exists')
+
+    if current_state == "present":
+        rc = instance.destroy()
+        if rc:
+            changed = True
+        else:
+            module.fail_json(
+                msg='instance template destroy failed'
+            )
+
+    json_data = {}
+    return (changed, json_data, name)
+
+
+def module_controller(module, gce):
+    ''' Control module state parameter.
+    module : AnsibleModule object
+    gce: authenticated GCE libcloud driver
+    Returns:
+        nothing
+    Exit:
+        AnsibleModule object exit with json data.
+    '''
+    json_output = dict()
+    state = module.params.get("state")
+    if state == "present":
+        (changed, output, name) = create_instance_template(module, gce)
+        json_output['changed'] = changed
+        json_output['msg'] = output
+    elif state == "absent":
+        (changed, output, name) = delete_instance_template(module, gce)
+        json_output['changed'] = changed
+        json_output['msg'] = output
+    else:
+        module.fail_json(
+            msg='bad state !',
+        )
+    module.exit_json(**json_output)
+
+
+def check_if_system_state_would_be_changed(module, gce):
+    ''' check_if_system_state_would_be_changed !
+    module : AnsibleModule object
+    gce: authenticated GCE libcloud driver
+    Returns:
+        system_state changed
+    '''
+    changed = False
+    current_state = "absent"
+
+    state = module.params.get("state")
+    name = module.params.get("name")
+
+    instance = None
+    try:
+        instance = gce.ex_get_instancetemplate(name)
+        current_state = "present"
+    except GoogleBaseError as err:
+        module.fail_json(msg='GCE get instancetemplate problem')
+
+    if current_state != state:
+        changed = True
+
+    if current_state == "absent":
+        if changed:
+            output = 'instance template {} will be created'.format(name)
+        else:
+            output = 'nothing to do for instance template {} '.format(name)
+    if current_state == "present":
+        if changed:
+            output = 'instance template {} will be detroyed'.format(name)
+        else:
+            output = 'nothing to do for instance template {} '.format(name)
+
+    return (changed, output)
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            state=dict(choices=['present', 'absent'], default='present'),
+            name=dict(require=True, aliases=['base_name']),
+            size=dict(default='f1-micro'),
+            source=dict(),
```

Comment:
In the docs you mention that source and image cannot both be applied. Can you enforce this by adding an argument like this: `mutually_exclusive=[['source', 'image']],` around line 555?